crates/matrix-sdk-ui/src/timeline/controller/observable_items.rs

```diff
@@ -1947,7 +1947,16 @@ impl AllRemoteEvents {
 
     /// Get the position of an event in the events array by its ID.
     pub fn position_by_event_id(&self, event_id: &EventId) -> Option<usize> {
-        self.0.iter().position(|event_meta| event_meta.event_id == event_id)
+        // Reverse the iterator to start looking at the end. Since this will give us the
+        // "reverse" position, reverse the index after finding the event.
+        //
+        // If the reversed index was 0 in a array of size 1, then it should stay 0, so
+        // that justifies the `- 1` in the formula below.
+        self.0
+            .iter()
+            .rev()
+            .position(|event_meta| event_meta.event_id == event_id)
+            .map(|rev_index| self.0.len() - 1 - rev_index)
```

Comment:
Could be simplified as:

```rust
self.0
    .iter()
    .enumerate()
    .rev()
    .find_map(|(index, event_meta)| { (event_meta.event_id == event_id).then_some(index))
```

So that you don't need to do the math yourself.

It works because [`Enumerate` implements `DoubleEndedIterator`](https://doc.rust-lang.org/std/iter/struct.Enumerate.html#impl-DoubleEndedIterator-for-Enumerate%3CI%3E).

Reply:
Good idea, thanks!